From c525e2783db452f86787f420e171c89502c9b1de Mon Sep 17 00:00:00 2001
From: Phil Sutter <psutter@redhat.com>
Date: Fri, 17 Mar 2017 13:17:53 +0100
Subject: [PATCH] tc: improve filter help texts a bit

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1422629
Upstream Status: iproute2.git commit 0a83e1eaf7b3c
Conflicts: Context changes due to missing commits 863ecb04b4f7b
	   ("discourage use of direct policer interface") and
	   287bf3a9900d1 ("route classifier support for multiple
	   actions")

commit 0a83e1eaf7b3c2ed9e6d809bd7aac7946f9d8d87
Author: Phil Sutter <phil@nwl.cc>
Date:   Fri Oct 23 19:21:17 2015 +0200

    tc: improve filter help texts a bit

    This fixes a few syntax errors and changes route filter help text to use
    classid instead of flowid to be consistent with other filters' help
    texts.

    Signed-off-by: Phil Sutter <phil@nwl.cc>
---
 tc/f_flower.c | 4 ++--
 tc/f_route.c  | 2 +-
 tc/f_rsvp.c   | 2 +-
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/tc/f_flower.c b/tc/f_flower.c
index 9a5ea06..a9b2c4d 100644
--- a/tc/f_flower.c
+++ b/tc/f_flower.c
@@ -28,7 +28,7 @@ static void explain(void)
 	fprintf(stderr, "                  [ action ACTION-SPEC ] [ classid CLASSID ]\n");
 	fprintf(stderr, "\n");
 	fprintf(stderr, "Where: MATCH-LIST := [ MATCH-LIST ] MATCH\n");
-	fprintf(stderr, "       MATCH      := [ indev DEV-NAME | \n");
+	fprintf(stderr, "       MATCH      := { indev DEV-NAME | \n");
 	fprintf(stderr, "                       dst_mac MAC-ADDR | \n");
 	fprintf(stderr, "                       src_mac MAC-ADDR | \n");
 	fprintf(stderr, "                       eth_type [ipv4 | ipv6 | ETH-TYPE ] | \n");
@@ -36,7 +36,7 @@ static void explain(void)
 	fprintf(stderr, "                       dst_ip [ IPV4-ADDR | IPV6-ADDR ] | \n");
 	fprintf(stderr, "                       src_ip [ IPV4-ADDR | IPV6-ADDR ] | \n");
 	fprintf(stderr, "                       dst_port PORT-NUMBER | \n");
-	fprintf(stderr, "                       src_port PORT-NUMBER | \n");
+	fprintf(stderr, "                       src_port PORT-NUMBER }\n");
 	fprintf(stderr,	"       FILTERID := X:Y:Z\n");
 	fprintf(stderr,	"       ACTION-SPEC := ... look at individual actions\n");
 	fprintf(stderr,	"\n");
diff --git a/tc/f_route.c b/tc/f_route.c
index 649e0ec..44361eb 100644
--- a/tc/f_route.c
+++ b/tc/f_route.c
@@ -28,7 +28,7 @@
 static void explain(void)
 {
 	fprintf(stderr, "Usage: ... route [ from REALM | fromif TAG ] [ to REALM ]\n");
-	fprintf(stderr, "                [ flowid CLASSID ] [ police POLICE_SPEC ]\n");
+	fprintf(stderr, "                [ classid CLASSID ] [ police POLICE_SPEC ]\n");
 	fprintf(stderr, "       POLICE_SPEC := ... look at TBF\n");
 	fprintf(stderr, "       CLASSID := X:Y\n");
 	fprintf(stderr, "\nNOTE: CLASSID is parsed as hexadecimal input.\n");
diff --git a/tc/f_rsvp.c b/tc/f_rsvp.c
index 44d3bdf..71508e1 100644
--- a/tc/f_rsvp.c
+++ b/tc/f_rsvp.c
@@ -27,7 +27,7 @@
 static void explain(void)
 {
 	fprintf(stderr, "Usage: ... rsvp ipproto PROTOCOL session DST[/PORT | GPI ]\n");
-	fprintf(stderr, "                [ sender SRC[/PORT | GPI ]\n");
+	fprintf(stderr, "                [ sender SRC[/PORT | GPI ] ]\n");
 	fprintf(stderr, "                [ classid CLASSID ] [ police POLICE_SPEC ]\n");
 	fprintf(stderr, "                [ tunnelid ID ] [ tunnel ID skip NUMBER ]\n");
 	fprintf(stderr, "Where: GPI := { flowlabel NUMBER | spi/ah SPI | spi/esp SPI |\n");
-- 
1.8.3.1

